src/store/getters.ts

```diff
@@ -9,3 +32,355 @@ export const getNameRequestUrl = (state: any): string => {
+export const getLocation = (state: StateIF): string => {
+  return state.stateModel.newRequestModel.location
+}
+
+/** Returns the location. */
+export const getEntityTypeCd = (state: StateIF): string => {
+  return state.stateModel.newRequestModel.entity_type_cd
+}
+
+/** Returns the current action code. */
+export const getRequestActionCd = (state: StateIF): string => {
+  return state.stateModel.newRequestModel.request_action_cd
+}
+
+/** Returns the current request jurisdiction code. */
+export const getRequestJurisdictionCd = (state: StateIF): string => {
+  return state.stateModel.newRequestModel.request_jurisdiction_cd
+}
+
+/** Returns the original request action code. */
+export const getRequestActionOriginal = (state: StateIF): string => {
+  return state.stateModel.newRequestModel.requestActionOriginal
+}
+
+/** Returns the applicant info. */
+export const getApplicant = (state: StateIF): ApplicantI => {
+  return state.stateModel.newRequestModel.applicant
+}
+
+/** Returns the corp seracho. */
+export const getCorpSearch = (state: StateIF): string => {
+  return state.stateModel.newRequestModel.corpSearch
+}
+
+/** Returns the analyze pending boolean. */
+export const getAnalyzePending = (state: StateIF): boolean => {
+  return state.stateModel.newRequestModel.analyzePending
+}
+
+/** Return the existing request data. */
+export const getExistingRequestSearch = (state: StateIF): ExistingRequestSearchI => {
+  return state.stateModel.newRequestModel.existingRequestSearch
+}
+
+/** Return the current quick search flag. */
+export const getQuickSearch = (state: StateIF): boolean => {
+  return state.stateModel.newRequestModel.quickSearch
+}
+
+/** Return the current errors. */
+export const getErrors = (state: StateIF): string[] => {
+  return state.stateModel.newRequestModel.errors
+}
+
+/** Return true if XproMras. */
+export const getIsXproMras = (state: StateIF): boolean => {
+  return (['CA', 'IN'].includes(getLocation(state)) && getRequestActionCd(state) !== 'MVE')
+}
+
+/** Return noCorpNum flag. */
+export const getHasNoCorpNum = (state: StateIF): boolean => {
+  return state.stateModel.newRequestModel.noCorpNum
+}
+
+/** Return english name flag. */
+export const getNameIsEnglish = (state: StateIF): boolean => {
+  return state.stateModel.newRequestModel.nameIsEnglish
+}
+
+/** Return persons name flag. */
+export const getIsPersonsName = (state: StateIF): boolean => {
+  return state.stateModel.newRequestModel.isPersonsName
+}
+
+/** Return the array of entities to NOT analyze. */
+export const getDoNotAnalyzeEntities = (state: StateIF): string[] => {
+  return state.stateModel.newRequestModel.doNotAnalyzeEntities
+}
+
+// /** Return the cancelled analysis boolean. */
+// export const getUserCancelledAnalysis = (state: StateIF): boolean => {
+//   return state.stateModel.newRequestModel.userCancelledAnalysis
+// }
+
+/** Return the BC Entity Types. */
+export const getEntityTypesBC = (state: StateIF): Array<EntityI> => {
```

Comment:
For future: theoretically getters are cached (as long as nothing they depend on, of perhaps `state` doesn't change).

Does this create a static list of entity types in BC? If so, maybe this should be done up-front, once, somewhere else.

Reply:
Ditto the other complex getters below.

Reply:
I know 😩  
The complex getters are 🤢 
I believe*, this list is dynamic based on the conditions in the Getter, so I'm not so sure it can be done once up front. 
These will be looked at further though!

Reply:
OK, put it on your long-term todo list.

I'll RC you my list FYI.

Reply:
Or maybe a getter isn't the right pattern. Maybe we should these lists once, up-front, when the NR type is set/changed.

Medium-term future?